src/quick-lint-js/fe/parse.cpp

```diff
@@ -322,10 +322,14 @@ void parser::error_on_pointless_string_compare(
   auto char_is_a_quote = [](const char8* s) {
     return *s == '"' || *s == '\'' || *s == '`';
   };
+  auto get_child_without_paren = [](expression* child) -> expression* {
+    return child->kind() == expression_kind::paren ? child->without_paren()
+                                                   : child;
+  };
```

Comment:
You don't need to check if the child is a paren before calling child->without_paren. Delete this function and just use without_paren directly.